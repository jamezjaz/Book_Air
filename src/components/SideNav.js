import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import logout from '../apiRequests/logoutRequest';

const SideNav = props => {
  const handleLogout = () => {
    const { signOut, user, loginStatus } = props;
    signOut();
    console.log(user);
    console.log(loginStatus);
  };

  return (
    <div>
      <button type="button" onClick={handleLogout}>Logout</button>
    </div>
  );
};

SideNav.propTypes = {
  signOut: PropTypes.func.isRequired,
  user: PropTypes.instanceOf(Object).isRequired,
  loginStatus: PropTypes.bool,
};

SideNav.defaultProps = {
  loginStatus: null,
};

const mapStateToProps = state => ({
  user: state.user,
  loginStatus: state.loginStatus,
});

const mapDispatchToprops = dispatch => ({
  signOut: user => dispatch(logout(user)),
});

export default connect(mapStateToProps, mapDispatchToprops)(SideNav);
