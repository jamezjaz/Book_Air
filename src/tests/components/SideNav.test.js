import React from 'react';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import SideNav from '../../components/SideNav';

import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();
const store = mockStore({});


describe('<SideNav />', () => {
  test('matches SideNav snapshot', () => {
    const tree = renderer
      .create(
      (<Provider store={store}>
        <Router>
          <SideNav />
        </Router>
      </Provider>),
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('renders SideNav component correctly', () => {
    render(
      <Provider store={store}>
        <Router>
          <SideNav />
        </Router>
      </Provider>,
    );
    expect(screen.getByText('AIRLINES')).toBeInTheDocument();
  });
});