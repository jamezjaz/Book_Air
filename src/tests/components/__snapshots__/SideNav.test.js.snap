// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideNav /> matches SideNav snapshot 1`] = `
<div
  className="container-fluid"
>
  <button
    className="rounded border-0"
    type="button"
  >
    <img
      alt="logo"
      src="paper-airplane.png"
    />
  </button>
  <div
    className="links"
  >
    <a
      aria-current={null}
      href="/airlinesList"
      onClick={[Function]}
    >
      AIRLINES
    </a>
    <a
      aria-current={null}
      href="/tickets"
      onClick={[Function]}
    >
      TICKETS
    </a>
    <a
      aria-current={null}
      href="/others"
      onClick={[Function]}
    >
      OTHERS
    </a>
  </div>
  <button
    className="logout"
    onClick={[Function]}
    type="submit"
  >
    Logout
  </button>
</div>
`;
